lib/Conversion/TritonGPUToLLVM/Utility.cpp

```diff
@@ -260,6 +260,16 @@ static Value commonShflSync(Location loc, ConversionPatternRewriter &rewriter,
                             const std::string &clamp, Value laneId = Value()) {
   unsigned bits = val.getType().getIntOrFloatBitWidth();
 
+#ifdef USE_ROCM
+  //On AMD, the ds_swizzle_b32 and ds_permute_b32 instructions work on 32bit/dwords
+  //so we need promote to 32 here.
+  if (bits == 8) {
+    Value i32Val = sext(i32_ty, val);
+    Value result = commonShflSync(loc, rewriter, i32Val, i, shuffleType, clamp, laneId);
+    return trunc(i8_ty, result);
```

Comment:
> is this related to the swizzle in GEMM and we need a test?

no it should be fine with GEMM, we compute swizzled addresses to read tensor data from LDS
But it could fire in FA, if we apply reduce on fp8 tensors

Reply:
I think fp8 can be handled in another PR.